doc/Programming-guide/Key-platform-features/Data-collection/index.md

```diff
@@ -5,4 +5,1175 @@ permalink: /:path/
+it is also possible to create custom log appenders.
+
+## Confirm delivery option
+
+By design every Kaa client stores logs in a log storage before sending them to the Kaa node. By default, upon receiving the logs, the corresponding
+log appender on the Kaa node sends the delivery confirmation back to the Kaa client. If the delivery was successful, the Kaa client deletes the log copies
+from its local storage, otherwise - it receives an appropriate error code and either restarts the operation of log delivery or attempts to deliver
+the same data to a different Kaa node.
+
+The confirm delivery option adds to a log appender the capability to verify whether received logs have been written into actual storage
+before sending the confirmation to the Kaa client.
+
+To illustrate, let’s consider the following scenarios:
+
+1. Log appender A has been configured with the confirm delivery option enabled. The Kaa client will receive a message about successful logs delivery
+only after the Kaa node successfully receives and writes the logs into an external storage. If the Kaa node fails to write the logs into the storage,
+the Kaa client will send the logs again. If there are more than one log appender with the enabled confirm delivery option then the Kaa client will receive
+a message about successful logs delivery only after each log appender successfully writes its logs into the storage.
+2. Log appender A has been configured with the confirm delivery option disabled. The Kaa client will receive a message about successful logs delivery
+to the Kaa node but if the Kaa node fails to write the logs into an external storage system, the Kaa client will not be informed about the failure and
+will not attempt sending the logs again. As a result, the logs will be lost.
+3. Log appender A has been configured with the confirm delivery option enabled and log appender B with the disabled one. The Kaa client will receive
+a message about successful logs delivery as soon as log appender A confirms delivery. Any errors that might happen when log appender B writes its logs into
+the storage will not be taken into account.
+
+To summarize, the confirm delivery option allows you to have the guaranteed delivery of every log record to the external storage.
+
+Also, it is worth noting that by default an inmemory log storage is used on the Kaa client. This means that you can lose your undelivered data in case of
+an endpoint reset. To avoid this, use a persistent log storage to store all the data that was not yet confirmed as delivered.
+
+## Existing log appender implementations
+
+There are several default log appender implementations that are available out of the box for each Kaa installation. A Kaa developer is able to add, update
+and delete log appenders using Admin UI or Admin REST API. After adding log appender please [configure your application](#data-collection-sdk-api) or use
```

Comment:
Could this be re-written to something like below?

After adding log appender ensure the data collection feature is [configured appropriately](#data-collection-sdk-api). For sample purposes [data collection demo](#data-collection-demo) application can be used as well.